src/document.c

```diff
@@ -593,6 +597,12 @@ FIELD_BULK_INDEXER(numericIndexer) {
       ctx->spec->stats.numRecords += rv.numRecords;
     }
   }
+  bool isGeo = (fs->types & INDEXFLD_T_GEO) != 0;
```

Comment:
this made me wonder rather we should have the metric update in one location where we iterate over the field types
Document_AddToIndexes for example and use the hash map to detect the field type like in FieldsGlobalStats_UpdateIndexError
because currently it feels kinda random and hard to maintain.
For example, text fields are updated in the preprocessor and numeric in the indexer. It's inconsistent